src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -37,8 +37,14 @@ public class Frame<TContext> : Frame
                     {
                         if (SocketInput.RemoteIntakeFin)
                         {
+                            if (TakeStartLine(SocketInput))
```

Comment:
The PR description describes very well why this is being done, but it's lost in the code. Please add a comment here explaining why this is needed.